From 9cb17d9fdffabab3693f6548f6bfc54ed2251325 Mon Sep 17 00:00:00 2001
From: Maggie Dreyer <maggie@puppet.com>
Date: Tue, 2 May 2017 10:21:06 -0700
Subject: [PATCH] (FACT-1624) Use `is_integer` instead of `is_fixednum` and
 `is_bignum`

Ruby 2.4 unified Fixnum and Bignum into Integer. We updated the
Leatherman API to match this. This commit updates Facter to use the new
API.
---
 lib/src/ruby/module.cc     | 2 +-
 lib/src/ruby/ruby_value.cc | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/lib/src/ruby/module.cc b/lib/src/ruby/module.cc
index 2d7223e0f..b8aeeed09 100644
--- a/lib/src/ruby/module.cc
+++ b/lib/src/ruby/module.cc
@@ -866,7 +866,7 @@ namespace facter { namespace ruby {
             // Unfortunately we have to call to_sym rather than using ID2SYM, which is Ruby version dependent
             uint32_t timeout = 0;
             volatile VALUE timeout_option = ruby.rb_hash_lookup(argv[1], ruby.to_symbol("timeout"));
-            if (ruby.is_fixednum(timeout_option)) {
+            if (ruby.is_integer(timeout_option)) {
                 timeout = ruby.num2size_t(timeout_option);
             }
 
diff --git a/lib/src/ruby/ruby_value.cc b/lib/src/ruby/ruby_value.cc
index d8ea20ce6..645472a3f 100644
--- a/lib/src/ruby/ruby_value.cc
+++ b/lib/src/ruby/ruby_value.cc
@@ -86,7 +86,7 @@ namespace facter { namespace ruby {
             json.SetString(str, size, allocator);
             return;
         }
-        if (ruby.is_fixednum(value) || ruby.is_bignum(value)) {
+        if (ruby.is_integer(value)) {
             json.SetInt64(ruby.rb_num2ll(value));
             return;
         }
@@ -155,7 +155,7 @@ namespace facter { namespace ruby {
             }
             return;
         }
-        if (ruby.is_fixednum(value) || ruby.is_bignum(value)) {
+        if (ruby.is_integer(value)) {
             os << ruby.rb_num2ll(value);
             return;
         }
@@ -241,7 +241,7 @@ namespace facter { namespace ruby {
             emitter << str;
             return;
         }
-        if (ruby.is_fixednum(value) || ruby.is_bignum(value)) {
+        if (ruby.is_integer(value)) {
             emitter << ruby.rb_num2ll(value);
             return;
         }
